internal/btf/co_re.go

```diff
@@ -0,0 +1,761 @@
+package btf
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/cilium/ebpf/internal"
+)
+
+var (
+	// ErrInvalidCORESpec indicates an invalid CO-RE relocation spec was parsed
+	ErrInvalidCORESpec        = errors.New("invalid CO-RE spec")
+	errStructureNeedsCleaning = errors.New("structure needs cleaning")
+	errUnsupportedRelocation  = errors.New("unsupported CO-RE relocation")
+)
+
+// LoadCORERelocations parses the relocations from the BTF extended ELF section and calculates how to apply
+// the relocation to the target, described by the BTF provided. If no BTF is provided, the current kernel BTF
+// will attempted to be loaded. Returns a mapping between instruction offset and the relocation, with result.
+func (s *Spec) LoadCORERelocations(name string, targetBtf *Spec, bo binary.ByteOrder) (map[uint64]*CORERelocation, error) {
```

Comment:
Can you add a check here for `s.byteOrder == target.byteOrder`? Then we can use `s.byteOrder` for the endianness check for bitfield.

Also, seems like `bo` is unused?